pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -226,6 +226,13 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 			ForwardPayloadHeader: jwtRule.OutputPayloadToHeader,
 			PayloadInMetadata:    jwtRule.Issuer,
 		}
+		if useExtendedJwt {
+			provider.PayloadInMetadata = filters.EnvoyJwtFilterPayload
```

Comment:
this would break a user who is already assuming Issuer  in the payload and using that for some reason?

Reply:
You mean the internal dynamic metadata from istio_authn? That was not a contract we generally follow - most of Envoy internal interfaces are private for a good reason.

Reply:
I mean any user can have a custom envoy filter that is reading the dynamic metadata from jwt_authn filter. It could break them?

Reply:
previously we were just adding issuer now `payload`. Wouldn't it be a problem?

Reply:
Yes, if they have a custom Envoy/Wasm filter, they need to update. That's always the expectation that the low-level Envoy interfaces are not stable; otherwise, we could never make Envoy changes. I'll update the change note to warn against this. It's a fairly simple changes.

Reply:
sounds good!!